plugins/meta/sbr/main.go

```diff
@@ -0,0 +1,291 @@
+	// chained).
+	// If you need to modify the result before returning it, you will need
+	// to actually convert it to a concrete versioned struct.
+	RawPrevResult *map[string]interface{} `json:"prevResult"`
+	PrevResult    *current.Result         `json:"-"`
+
+	// Add plugin-specific flags here
+	LogFile           string `json:"logfile"`
+}
+
+// parseConfig parses the supplied configuration (and prevResult) from stdin.
+func parseConfig(stdin []byte) (*PluginConf, error) {
+	conf := PluginConf{}
+
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse network configuration: %v", err)
+	}
+
+	// Parse previous result.
+	if conf.RawPrevResult != nil {
+		resultBytes, err := json.Marshal(conf.RawPrevResult)
+		if err != nil {
+			return nil, fmt.Errorf("could not serialize prevResult: %v", err)
+		}
+		res, err := version.NewResult(conf.CNIVersion, resultBytes)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse prevResult: %v", err)
+		}
+		conf.RawPrevResult = nil
+		conf.PrevResult, err = current.NewResultFromResult(res)
+		if err != nil {
+			return nil, fmt.Errorf("could not convert result to current version: %v", err)
+		}
+	}
+	// End previous result parsing
+
+	if conf.LogFile != "" {
+		file, err := os.OpenFile(conf.LogFile, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0666)
+		if err != nil {
+			return nil, err
+		}
+
+		log.SetOutput(file)
+	}
+
+	return &conf, nil
+}
+
+// cmdAdd is called for ADD requests
+func cmdAdd(args *skel.CmdArgs) error {
+	conf, err := parseConfig(args.StdinData)
+	if err != nil {
+		return err
+	}
+
+	if conf.PrevResult == nil {
+		return fmt.Errorf("This plugin must be called as chained plugin")
+	}
+
+	// We do a single interface name, stored in args.IfName
+	log.Printf("Working with interface: %s", args.IfName)
+
+	// Get list of IP addresses on that interface
+	if len(conf.PrevResult.IPs) == 0 {
+		// No IP addresses; that makes no sense. Pack it in.
+		return fmt.Errorf("No IP addresses on interface: %s", args.IfName)
+	}
+
+	// ips contains the IPConfig structures that were passed, filtered somewhat
+	ips := make([]*current.IPConfig, 0, len(conf.PrevResult.IPs))
+
+	for _, ip := range conf.PrevResult.IPs {
+		// IPs have an interface that is an index into the interfaces array.
+		if ip.Interface == nil {
+			log.Printf("No interface for IP address %s", ip.Address.IP)
+			continue
+		}
+
+		// Skip all IPs we know belong to an interface with the wrong name.
+		intIdx := *ip.Interface
+		if intIdx >= 0 && intIdx < len(conf.PrevResult.Interfaces) && conf.PrevResult.Interfaces[intIdx].Name != args.IfName {
+			log.Printf("Incorrect interface for IP address %s", ip.Address.IP)
+			continue
+		}
+
+		log.Printf("Found IP address %s", ip.Address.IP.String())
+		ips = append(ips, ip)
+	}
+
+	nsname := os.Getenv("CNI_NETNS")
+	log.Printf("Network namespace to use: %s", nsname)
+	namespace, err := ns.GetNS(nsname)
```

Comment:
Can simplify this with ns.WithNetNSPath() rather than getting the namespace object and then Do() later.